static/js/lib/ckeditor/plugins/Markdown.test.ts

```diff
@@ -30,3 +26,71 @@ describe("Markdown CKEditor plugin", () => {
     })
   })
 })
+
+describe("Multiple Markdown CKEditors", () => {
+  /**
+   * A markdown plugin that inteprets all HTML paragraphs as a particular animal
+   * sound...
+   *
+   * The goal here is to construct two conflicting plugins (say, CatPlugin and
+   * DogPlugin) and demonstrate two separate editors can each independently use
+   * one of the conflicting plugins.
+   */
+  abstract class AnimalParagraphSyntax extends MarkdownSyntaxPlugin {
+    abstract sound: string
+    abstract pluginName: string
+
+    get showdownExtension(): () => ShowdownExtension[] {
+      return () => []
+    }
+
+    get turndownRules(): TurndownRule[] {
+      return [
+        {
+          name: this.plugin_name,
+          rule: {
+            filter:      "p",
+            replacement: content => {
+              const sound = this.sound
+              const sounds = sound.repeat(
+                Math.ceil(content.length / sound.length)
+              )
+              return `${sounds}!`
+            }
+          }
+        }
+      ]
+    }
+  }
+
+  /**
+   * Meow!
+   */
+  class CatParagraphSyntax extends AnimalParagraphSyntax {
+    pluginName = "cat_paragraphs"
+    sound = "meow"
+  }
+
+  /**
+   * Woof!
+   */
+  class DogParagraphSyntax extends AnimalParagraphSyntax {
+    pluginName = "dog_paragraphs"
+    sound = "woof"
+  }
+
+  const getCatEditor = createTestEditor([CatParagraphSyntax, Markdown])
+  const getDogEditor = createTestEditor([DogParagraphSyntax, Markdown])
+
+  test("Separate editors can use conflciting turndown plugins", async () => {
+    const catEditor = await getCatEditor()
+    const dogEditor = await getDogEditor()
+    const cat = getConverters(catEditor)
+    const dog = getConverters(dogEditor)
+
+    const paragraph = "<p>1234567890</p>"
+
+    expect(cat.html2md(paragraph)).toBe("meowmeowmeow!")
+    expect(dog.html2md(paragraph)).toBe("woofwoofwoof!")
```

Comment:
This test would fail on master. The dog markdown would be "meowmeowmeow!". Poor dog.